src/api/schemas/business-schema.ts

```diff
@@ -0,0 +1,121 @@
+import { object, string } from 'zod';
+
+const employeeSchema = object({
+    firebase_id: string({
+        required_error: 'Firebase ID is required',
+        invalid_type_error: 'Firebase ID should be of type string'
+    }),
+    firstName: string({
+        required_error: 'First Name is required',
+        invalid_type_error: 'First Name should be of type string'
+    }).min(2, 'First Name too short!'),
+    lastName: string({
+        required_error: 'Last Name is required',
+        invalid_type_error: 'Last Name should be of type string'
+    }).min(2, 'Last Name too short!'),
+    email: string({
+        required_error: 'Email is required',
+        invalid_type_error: 'Email should be of type string'
+    }).email('Invalid email'),
+    role: string({
+        required_error: 'Role is required',
+        invalid_type_error: 'Role should be of type string'
+    })
+});
+
+const addressSchema = object({
+    // Required
+    lineOne: string({
+        required_error: 'Line One is required',
+        invalid_type_error: 'Line One should be of type string'
+    }).min(5, 'Line One too short!'),
+    city: string({
+        required_error: 'City is required',
+        invalid_type_error: 'City should be of type string'
+    }),
+    state: string({
+        required_error: 'State/Province is required',
+        invalid_type_error: 'State/Province should be of type string'
+    }).regex(new RegExp('^[A-Z]{2,4}')),
+    country: string({
+        required_error: 'Country is required',
+        invalid_type_error: 'Country should be of type string'
+    }),
+    postalCode: string({
+        required_error: 'Postal code is required',
+        invalid_type_error: 'Postal Code should be of type string'
+    }),
+
+    // Optional
+    mention: string({
+        invalid_type_error: 'Mention should be of type string'
+    }).optional(),
+    lineTwo: string({
+        invalid_type_error: 'Line Two should be of type string'
+    }).optional()
+});
+
+const getBusinessById = object({
+    params: object({
+        businessId: string({
+            required_error: 'Business ID is required',
+            invalid_type_error: 'Business ID should be of type string'
+        })
+    }).strict('Request URL contains an invalid key')
```

Comment:
If I remove business ID from the params it correctly return found: "false" but should be intercepted by validator first. I wonder why its not since you did enforce .strict on its params object...